### PR TITLE
[POC] [WIP] G-API: Introduce mechanism to overcome backends testing challenges

### PR DESCRIPTION
**Pull request by andrey-golubev**
### This pullrequest

is a prototype to showcase how we can reduce (potentially by much) the test code boilerplate

### CODE REVIEW

**Comment by andrey-golubev** [comment]
This thing alone is already -4 LOC compared to master OpenCV

File: modules/gapi/test/cpu/gapi_core_tests_cpu.cpp [hunk 1]
```diff
@@ -16,21 +17,28 @@ namespace
...
+                                Values(CompileArgs), \
+                                Values(ADD, MUL), \
+                                testing::Bool(), \
+                                Values(1.0), \
+                                Values(false)), \
+                        opencv_test::PrintMathOpCoreParams()); \
+
+// Now we have all 3 backends tested with the same arguments with a one-liner
+// FIXME: same arguments is not always the case
+REGISTER_FOR_ALL(INSTANTIATE_MATH_OP_1)
```
Comment (dmatveev) [inline modules/gapi/test/cpu/gapi_core_tests_cpu.cpp]: No macros like this, please.

It increases coupling between tests for different backends.

**Comment by andrey-golubev** [comment]
Ok, I see, wasn't sure myself if such approach is a good thing. Let's close this one then

**Comment by dmatveev** [comment]
Thanks for a try anyway!

**Pull request closed without merging**